fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -408,6 +405,10 @@ def reduce(*_: Any) -> None:
                 if not self.should_accumulate_grads and self._grad_to_be_reduced[index]:
                     assert param.grad is not None, "Reducing gradients during backward pass, cannot be None"
 
+                    if not self._bucket_flush_callback_set:
```

Comment:
some context: the trick is that you have to be within the BW pass to be able to add this flush callback. So it's re-added for every BW pass, one of the existing callbacks checks whether it's there, if not added to the queue. The callback ordering is unknown, so there's not other way that I know of